### PR TITLE
UsageTracker: process created series events

### PR DESCRIPTION
**Pull request by colega**
This adds a method to process series creation events, also a test that checks that it's compatible with the events produced.

### CODE REVIEW

File: pkg/usagetracker/tracker_store.go [hunk 1]
```diff
@@ -300,16 +341,54 @@ func (shard *tenantShard) trackSeries(series []uint64, now minutes, totalTenantS
...
-	rejected = append(rejected, pending...)
+	// Whatever is still missing is rejected (if any)
+	rejected = append(rejected, missing...)
 	return created, rejected
 }
 
+// processCreatedSeriesEvent creates the series coming from an event from Kafka,
+// i.e. series that were created by a different instance.
+// This does not check the limits, as we prioritize staying consistent across replicas over enforcing limits.
+// Timestamp is not updated if series exists already.
```
Comment (pracucci) [inline pkg/usagetracker/tracker_store.go:357]: Shouldn't be updated with the highest value between the two? I guess it shouldn't happen, but never know.

File: pkg/usagetracker/tracker_store.go [hunk 2]
```diff
@@ -300,16 +341,54 @@ func (shard *tenantShard) trackSeries(series []uint64, now minutes, totalTenantS
...
+func (shard *tenantShard) processCreatedSeriesEvent(series []uint64, timestamp minutes, totalTenantSeries *atomic.Uint64) {
+	// missing contains the series that have to be created.
+	// As we advance through series, we reuse the same slice to avoid allocations.
+	missing := series[:0]
+
+	// First try to update the series that already exist, the ones that don't exist are moved to missing.
```
Comment (pracucci) [inline pkg/usagetracker/tracker_store.go:363]: [nit] The comment says "update" but we're not doing any update here. We're just looking for missing ones.

**Review by pracucci — APPROVED** [review]
Nice work! This was easy to review.

File: pkg/usagetracker/tracker_store.go (hunk 1 above)
Comment (colega) [reply pkg/usagetracker/tracker_store.go:357]: It could happen but we don't really care as we'll update when we process the snapshot later.

**Pull request merged**